app/processing/account-codes/map-account-codes.js

```diff
@@ -13,8 +13,8 @@ const mapAccountCodes = async (paymentRequest) => {
 
   for (const invoiceLine of paymentRequest.invoiceLines) {
     const lineCode = getLineCodeFromDescription(invoiceLine.description)
-    const accountCodesForLine = getCodesForLine(paymentRequest.schemeId, lineCode, invoiceLine.schemeCode, accountCodeMap)
-    invoiceLine.accountCode = selectLineCode(accountCodesForLine, paymentRequest.ledger, paymentRequest.debtType)
+    const accountCodesForLine = getCodesForLine(paymentRequest.schemeId, lineCode, invoiceLine.schemeCode, invoiceLine.stateAid, accountCodeMap)
+    invoiceLine.accountCode = selectLineCode(accountCodesForLine, paymentRequest.ledger, paymentRequest.debtType, invoiceLine.stateAid, invoiceLine.accountCode)
   }
 }
```

Comment:
Yeah fair comment.  I've passed in the invoice line and payment request objects.  Wanted to avoid that but I think you're right, it's getting harder to see where values come from as is.